tools/operator/fabricconfiguration/configtx.go

```diff
@@ -0,0 +1,933 @@
+	signcerts            = "signcerts"
+	keystore             = "keystore"
+	intermediatecerts    = "intermediatecerts"
+	crlsfolder           = "crls"
+	configfilename       = "config.yaml"
+	tlscacerts           = "tlscacerts"
+	tlsintermediatecerts = "tlsintermediatecerts"
+)
+
+func convertToUint(size string) uint32 {
+	var sizeToInt int
+	size = strings.Trim(size, " MB") 
+	sizeToInt, _ = strconv.Atoi(size)
+	return uint32(sizeToInt * 1024 * 1024)
+}
+
+func doOutputBlock(config *genesisconfig.Profile, channelID string, outputBlock string) error {
+	channel := newChannel(config)
+	genesisBlock, err := configtx.NewSystemChannelGenesisBlock(channel, channelID)
+	if err != nil {
+		panic(err)
+	}
+	logger.Info("Writing genesis block")
+	err = writeFile(outputBlock, protoutil.MarshalOrPanic(genesisBlock), 0640)
+	if err != nil {
+		return fmt.Errorf("Error writing genesis block: %s", err)
+	}
+	return nil
+}
+
+func doOutputChannelCreateTx(conf *genesisconfig.Profile, channelID string, outputChannelCreateTx string) error {
+	logger.Info("Generating new channel configtx")
+
+	channel := newChannel(conf)
+	configUpdate, err := configtx.NewMarshaledCreateChannelTx(channel, channelID)
+	env, err := configtx.NewEnvelope(configUpdate)
+	if err != nil {
+		panic(err)
+	}
+
+	logger.Info("Writing new channel tx")
+	err = writeFile(outputChannelCreateTx, protoutil.MarshalOrPanic(env), 0640)
+	if err != nil {
+		return fmt.Errorf("Error writing channel create tx: %s", err)
+	}
+	return nil
+}
+
+func writeFile(filename string, data []byte, perm os.FileMode) error {
+	dirPath := filepath.Dir(filename)
+	exists, err := dirExists(dirPath)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		err = os.MkdirAll(dirPath, 0750)
+		if err != nil {
+			return err
+		}
+	}
+	return ioutil.WriteFile(filename, data, perm)
+}
+
+func dirExists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func newChannel(baseProfile *genesisconfig.Profile) configtx.Channel {
+	// Application section
+	application := configtx.Application{}
+	if baseProfile.Application != nil {
+		appOrgs := newOrganization(baseProfile.Application.Organizations)
+		appCapabilities := []string{}
+		for name := range baseProfile.Application.Capabilities {
+			appCapabilities = append(appCapabilities, name)
+		}
+		appPolicies := newPolicies(baseProfile.Application.Policies)
+		application = configtx.Application{
+			Organizations: appOrgs,
+			Capabilities:  appCapabilities,
+			Policies:      appPolicies,
+			ACLs:          baseProfile.Application.ACLs,
+		}
+	}
+
+	// Orderer section
+	o := configtx.Orderer{}
+	if baseProfile.Orderer != nil {
+		addresses := []configtx.Address{}
+		for _, addr := range baseProfile.Orderer.Addresses {
+			addrslice := strings.Split(addr, ":")
+			port, err := strconv.Atoi(addrslice[1])
+			if err != nil {
+				panic(err)
```

Comment:
Return error.

All places that panic should return an error instead

Reply:
Fixed